### PR TITLE
Prevent window from being closed when before-quit is default prevented on non MacOs

### PR DESCRIPTION
**Pull request by darkdh**
required by https://github.com/brave/browser-laptop/pull/10687

Auditors: @bridiver, @bsclifton, @bbondy

### CODE REVIEW

**Comment by bsclifton** [comment]
@darkdh do you have more information about this one?

ex: how can I test behavior before and after?

**Comment by darkdh** [comment]
@bsclifton, you can simply put `e.preventDefault()` in the beginning of our [`app.on('before-quit', [](e)=>...`](https://github.com/brave/browser-laptop/blob/master/app/sessionStoreShutdown.js#L173)
on Windows/Linux

`npm start` from `browser-laptop-bootstrap` built with my change
`npm start` from `browser-laptop-bootstrap/src/browser-laptop` before change so you don't have to rebuild

And when you quit Brave, you should be able to see the window is not closed

File: atom/browser/extensions/tab_helper.cc [hunk 1]
```diff
@@ -253,6 +253,9 @@ void TabHelper::WillCloseWindow(bool* prevent_default) {
     browser()->window()->Deactivate();
     browser()->window()->Hide();
   }
+  if (!browser_shutdown::IsTryingToQuit()) {
```
Comment (bridiver) [inline atom/browser/extensions/tab_helper.cc]: this can be handled by calling prevent default `app.on('close'` in browser-laptop

**Comment by bridiver** [comment]
I think the problem is in the original browser-laptop pr and I don't think any muon changes are necessary

**Comment by darkdh** [comment]
oh, yeah. Just see your comment
I will update it on the original issue. Thanks.

**Comment by bridiver** [comment]
I think the real problem is not with e.preventDefault in before-quit, but verifying now

**Pull request closed without merging**